stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -649,7 +649,40 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     ///
     /// The iterator will increment byte-by-byte.
     public func makeIterator() -> Data.Iterator {
-        return IndexingIterator(_elements: self)
+        return Iterator(_data: self)
+    }
+
+    public struct Iterator : IteratorProtocol {
+        private let _data: Data
+        private var _buffer: (
+            UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8,
+            UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8,
+            UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8,
+            UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8, UInt8)
+        private var _idx: Data.Index
+        private let _endIdx: Data.Index
+
+        private init(_data: Data) {
+            self._data = _data
+            _buffer = (0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0)
+            _idx = 0
+            _endIdx = _data.endIndex
+        }
+
+        public mutating func next() -> UInt8? {
+            guard _idx < _endIdx else { return nil }
+            defer { _idx += 1 }
+            let bufferSize = sizeofValue(_buffer)
+            return withUnsafeMutablePointer(to: &_buffer) { ptr_ in
+                let ptr = UnsafeMutablePointer<UInt8>(ptr_)
```

Comment:
That's right! You need to drop down to a raw pointer then restate the pointer type. It's weird that this needs to be done for tuples. If it's very common we'll need to come up with a better tuple-specific way.